### DISCUSSION TITLE
Looking for "pixel engine" optimisation suggestions.

### DISCUSSION THREAD

**Discussion by IsaacGaches** [bevyengine/bevy#6888] [Q&A]
I'm working on a 2d voxel engine (a pixel engine? like terraria or starbound). It has good performance however for the sake of learning and curiosity, I'd like to know every way to optimise.
If you know anything, at least just point me in the right direction please. For example: what's the best way to render blocks? or techniques like chunking and multithreading.
Thanks in advance.

**Comment by CatCode79** [comment]
I have no idea about the techniques personally, try checking out this sandbox game: https://github.com/JMS55/sandbox

I see the project uses [pixels](https://github.com/parasyte/pixels), there is a [Bevy plugin for it](https://github.com/dtcristo/bevy_pixels).

**Reply by IsaacGaches** [reply]
thanks, I'll have a further look
